xml/mysql_banners.xml

```diff
@@ -0,0 +1,850 @@
+    <param pos="0" name="service.family" value="MySQL"/>
+    <param pos="0" name="service.product" value="MySQL"/>
+    <param pos="0" name="service.edition" value="Community Edition"/>
+    <param pos="0" name="os.vendor" value="CloudLinux"/>
+    <param pos="0" name="os.device" value="General"/>
+    <param pos="0" name="os.family" value="Linux"/>
+    <param pos="0" name="os.product" value="Linux"/>
+  </fingerprint>
+
+  <fingerprint pattern="^(\d\.\d{1,2}\.\d{1,2})-0ubuntu0.(\d\d\.\d\d)">
+    <example>5.1.66-0ubuntu0.10.04.3-log</example>
+    <example>5.5.28-0ubuntu0.12.04.3-log</example>
+    <description>Oracle MySQL on Ubuntu</description>
+    <param pos="1" name="service.version"/>
+    <param pos="0" name="service.vendor" value="Oracle"/>
+    <param pos="0" name="service.family" value="MySQL"/>
+    <param pos="0" name="service.product" value="MySQL"/>
+    <param pos="0" name="os.vendor" value="Ubuntu"/>
+    <param pos="0" name="os.device" value="General"/>
+    <param pos="0" name="os.family" value="Linux"/>
+    <param pos="0" name="os.product" value="Linux"/>
+    <param pos="2" name="os.version"/>
+  </fingerprint>
+
+  <fingerprint pattern="^(\d\.\d{1,2}\.\d{1,2})-(?:Debian_)?\dubuntu(\d{1,2}\.\d\d)">
+    <example>5.0.22-Debian_0ubuntu6.06.14-log</example>
+    <example>5.1.41-3ubuntu12.10</example>
+    <description>Oracle MySQL on Ubuntu</description>
+    <param pos="1" name="service.version"/>
+    <param pos="0" name="service.vendor" value="Oracle"/>
+    <param pos="0" name="service.family" value="MySQL"/>
+    <param pos="0" name="service.product" value="MySQL"/>
+    <param pos="0" name="os.vendor" value="Ubuntu"/>
+    <param pos="0" name="os.device" value="General"/>
+    <param pos="0" name="os.family" value="Linux"/>
+    <param pos="0" name="os.product" value="Linux"/>
+    <param pos="2" name="os.version"/>
+  </fingerprint>
+
+  <fingerprint pattern="^(\d\.\d{1,2}\.\h{1,3})[-_](?:Ubuntu[_-])?(?:Debian[_-])?\d{1,2}(?:~exp1)?ubuntu\d{1,2}(?:\.\d)?(?:-log)?$">
+    <example service.version="5.0.45">5.0.45-Debian_1ubuntu3.4</example>
+    <example service.version="5.0.51a">5.0.51a-3ubuntu5.5</example>
+    <example service.version="5.0.51a">5.0.51a-3ubuntu5.8-log</example>
+    <example service.version="5.0.75">5.0.75-0ubuntu10.2</example>
+    <example service.version="5.1.58">5.1.58-1ubuntu1</example>
+    <example service.version="4.0.24">4.0.24_Debian-10ubuntu2-log</example>
+    <example service.version="5.6.19">5.6.19-1~exp1ubuntu2-log</example>
+    <description>Oracle MySQL on Ubuntu where trailing digits aren't OS version</description>
+    <param pos="1" name="service.version"/>
+    <param pos="0" name="service.vendor" value="Oracle"/>
+    <param pos="0" name="service.family" value="MySQL"/>
+    <param pos="0" name="service.product" value="MySQL"/>
+    <param pos="0" name="os.vendor" value="Ubuntu"/>
+    <param pos="0" name="os.device" value="General"/>
```

Comment:
os.device is totally valid, but IMO there is little value in using it in mysql fingerprints unless you could somehow determine something more specific.